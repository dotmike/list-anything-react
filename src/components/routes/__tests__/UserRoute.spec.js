import React from "react";
import configureStore from "redux-mock-store";

import ConnectedUserRoute, { UnconnectedUserRoute } from "../UserRoute";

const defaultProps = {
  component: () => <span>Testando...</span>
};

describe("ConnectedUserRoute", () => {
  const mockStore = configureStore();
  const initialState = {};
  const props = { ...defaultProps, store: mockStore(initialState) };

  it("renders correctly", () => {
    const wrapper = shallowWithContext(<ConnectedUserRoute {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("UnconnectedUserRoute", () => {
  const props = {
    ...defaultProps
  };

  describe("when `isAuthenticated` is false", () => {
    it("renders correctly", () => {
      props.isAuthenticated = false;
      const wrapper = mountWithContext(<UnconnectedUserRoute {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when `isAuthenticated` is true", () => {
    it("renders correctly", () => {
      props.isAuthenticated = true;
      const wrapper = mountWithContext(<UnconnectedUserRoute {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
