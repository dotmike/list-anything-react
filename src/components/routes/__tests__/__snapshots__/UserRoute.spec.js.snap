// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedUserRoute renders correctly 1`] = `
<UserRoute
  component={[Function]}
  dispatch={[Function]}
  isAuthenticated={false}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`UnconnectedUserRoute when \`isAuthenticated\` is false renders correctly 1`] = `
<UserRoute
  component={[Function]}
  isAuthenticated={false}
>
  <Route
    render={[Function]}
  >
    <Redirect
      push={false}
      to="/"
    />
  </Route>
</UserRoute>
`;

exports[`UnconnectedUserRoute when \`isAuthenticated\` is true renders correctly 1`] = `
<UserRoute
  component={[Function]}
  isAuthenticated={true}
>
  <Route
    render={[Function]}
  >
    <component
      history={
        Object {
          "_listeners": Array [],
          "_notifyListeners": [Function],
          "action": "POP",
          "createHref": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "111111",
            "pathname": "/",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      location={
        Object {
          "hash": "",
          "key": "111111",
          "pathname": "/",
          "search": "",
        }
      }
      match={
        Object {
          "isExact": true,
          "params": Object {},
          "path": "/",
          "url": "/",
        }
      }
    >
      <span>
        Testando...
      </span>
    </component>
  </Route>
</UserRoute>
`;
